relay_messages.go

```diff
@@ -249,6 +253,16 @@ func (f *lazyCallReq) Arg2Iterator() (arg2.KeyValIterator, error) {
 	return arg2.NewKeyValIterator(f.Payload[f.arg2StartOffset:f.arg2EndOffset])
 }
 
+func (f *lazyCallReq) Arg2Append(key, val []byte) {
+	// TODO(echung): pool the appends buffers
+	if len(f.arg2Appends) == cap(f.arg2Appends) {
+		newArg2Appends := make([]relay.KeyVal, len(f.arg2Appends), 2*cap(f.arg2Appends))
+		copy(newArg2Appends, f.arg2Appends)
+		f.arg2Appends = newArg2Appends
+	}
```

Comment:
if we use the array approach, we shouldn't need any of this. all we're doing is reimplementing slice logic here, we'd be better off just pooling the slice completely if we wanted to do this